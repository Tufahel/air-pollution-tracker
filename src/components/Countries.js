import React, { useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { getCountries } from '../redux/Actions/Country';
import Country from './Country';

const Countries = () => {
  const countries = useSelector((state) => state.countryReducer);
  const dispatch = useDispatch();

  useEffect(() => {
    dispatch(getCountries());
  }, []);
  return (
    <div>
      <Country
        key="1"
        id="1"
        name="name"
        region="region"
      />
      {
        countries.map((country) => (
          <Country
            key={country.code}
            id={country.code}
            name={country.name.common}
            region={country.region}
          />
        ))
      }
    </div>
  );
};

export default Countries;
