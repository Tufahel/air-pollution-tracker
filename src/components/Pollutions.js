import React from 'react';
import { useSelector } from 'react-redux';
import Header from './Header';
import Pollution from './Pollution';

const Pollutions = () => {
  const pollutions = useSelector((state) => state.pollutionReducer);

  return (
    <div>
      <Header id="/countries" />
      <Pollution
        key={pollutions.id}
        id={pollutions.id}
        lat={pollutions.lat}
        lng={pollutions.lng}
        co={pollutions.co}
        no={pollutions.no}
        no2={pollutions.no2}
        flag={pollutions.flag}
        name={pollutions.name}
      />
    </div>
  );
};

export default Pollutions;
