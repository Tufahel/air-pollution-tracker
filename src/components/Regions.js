import React from 'react';
import Header from './Header';
import Region from './Region';

const Regions = () => {
  const regionList = [
    { region: 'Asia', country: 50 },
    { region: 'Americas', country: 56 },
    { region: 'Oceania', country: 27 },
    { region: 'Antarctic', country: 5 },
    { region: 'Europe', country: 53 },
    { region: 'Africa', country: 59 }];
  // const regionCountry = ['50', '56', '27', '5', '53', '59'];
  return (
    <div>
      <Header id="/" />
      <div className="row">
        {
            regionList.map((element) => (
              <Region
                key={element.region}
                region={element.region}
                regionCountry={element.country}
              />
            ))
        }
      </div>
    </div>
  );
};

export default Regions;
