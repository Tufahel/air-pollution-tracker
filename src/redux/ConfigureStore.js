import {
  createStore, combineReducers, applyMiddleware, compose,
} from 'redux';
import logger from 'redux-logger';
import thunk from 'redux-thunk';
import countryReducer from './Reducers/Country';
import pollutionReducer from './Reducers/Pollution';

const rootReducer = combineReducers({
  countryReducer,
  pollutionReducer,
});

const store = createStore(rootReducer,
  compose(applyMiddleware(thunk, logger)));
export default store;
