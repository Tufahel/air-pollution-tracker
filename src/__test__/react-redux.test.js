import renderer from 'react-test-renderer';
import { cleanup } from '@testing-library/react';
import { BrowserRouter as Router } from 'react-router-dom';
import { Provider } from 'react-redux';
import { combineReducers, createStore, applyMiddleware } from 'redux';
import thunk from 'redux-thunk';
import countryReducer from '../redux/Reducers/Country';
import pollutionReducer from '../redux/Reducers/Pollution';
import Regions from '../components/Regions';
import Countries from '../components/Countries';
import Pollutions from '../components/Pollutions';

const rootReducer = combineReducers({
  countryReducer,
  pollutionReducer,
});
const store = createStore(rootReducer, applyMiddleware(thunk));

afterEach(cleanup);

describe('Region List', () => {
  test('Region Lists renders correctly', () => {
    const regions = renderer
      .create(
        <Provider store={store}>
          <Router>
            <Regions />
          </Router>
        </Provider>,
      )
      .toJSON();
    expect(regions).toMatchSnapshot();
  });
});

describe('Countries List ', () => {
  test('Countries List renders correctly', () => {
    const countries = renderer
      .create(
        <Provider store={store}>
          <Router>
            <Countries />
          </Router>
        </Provider>,
      )
      .toJSON();
    expect(countries).toMatchSnapshot();
  });
});

describe('Pollutions List ', () => {
  test('Pollution lists render correctly', () => {
    const pollutions = renderer
      .create(
        <Provider store={store}>
          <Router>
            <Pollutions />
          </Router>
        </Provider>,
      )
      .toJSON();
    expect(pollutions).toMatchSnapshot();
  });
});
