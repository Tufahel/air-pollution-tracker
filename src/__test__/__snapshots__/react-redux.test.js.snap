// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Countries List  Countries List renders correctly 1`] = `
<div
  className="m-2"
>
  <p>
    Please wait, Loading...
  </p>
  <a
    aria-current="page"
    className="link text-light active"
    href="/"
    onClick={[Function]}
  >
    <p>
      If not working, click to reload
    </p>
    <svg
      aria-hidden="true"
      className="svg-inline--fa fa-arrows-rotate icon"
      data-icon="arrows-rotate"
      data-prefix="fas"
      focusable="false"
      role="img"
      style={Object {}}
      text="reload"
      viewBox="0 0 512 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M464 16c-17.67 0-32 14.31-32 32v74.09C392.1 66.52 327.4 32 256 32C161.5 32 78.59 92.34 49.58 182.2c-5.438 16.81 3.797 34.88 20.61 40.28c16.89 5.5 34.88-3.812 40.3-20.59C130.9 138.5 189.4 96 256 96c50.5 0 96.26 24.55 124.4 64H336c-17.67 0-32 14.31-32 32s14.33 32 32 32h128c17.67 0 32-14.31 32-32V48C496 30.31 481.7 16 464 16zM441.8 289.6c-16.92-5.438-34.88 3.812-40.3 20.59C381.1 373.5 322.6 416 256 416c-50.5 0-96.25-24.55-124.4-64H176c17.67 0 32-14.31 32-32s-14.33-32-32-32h-128c-17.67 0-32 14.31-32 32v144c0 17.69 14.33 32 32 32s32-14.31 32-32v-74.09C119.9 445.5 184.6 480 255.1 480c94.45 0 177.4-60.34 206.4-150.2C467.9 313 458.6 294.1 441.8 289.6z"
        fill="currentColor"
        style={Object {}}
      />
    </svg>
  </a>
</div>
`;

exports[`Pollutions List  Pollution lists render correctly 1`] = `
<div>
  <div>
    <nav>
      <div
        className="d-flex justify-content-between align-items-center p-3 mb-2"
      >
        <button
          className="btn mt-2"
          type="button"
        >
          <a
            className="link m-2 text-light"
            href="/countries"
            onClick={[Function]}
          >
            <svg
              aria-hidden="true"
              className="svg-inline--fa fa-circle-arrow-left icon"
              data-icon="circle-arrow-left"
              data-prefix="fas"
              focusable="false"
              role="img"
              style={Object {}}
              viewBox="0 0 512 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M256 0C114.6 0 0 114.6 0 256c0 141.4 114.6 256 256 256s256-114.6 256-256C512 114.6 397.4 0 256 0zM384 288H205.3l49.38 49.38c12.5 12.5 12.5 32.75 0 45.25s-32.75 12.5-45.25 0L105.4 278.6C97.4 270.7 96 260.9 96 256c0-4.883 1.391-14.66 9.398-22.65l103.1-103.1c12.5-12.5 32.75-12.5 45.25 0s12.5 32.75 0 45.25L205.3 224H384c17.69 0 32 14.33 32 32S401.7 288 384 288z"
                fill="currentColor"
                style={Object {}}
              />
            </svg>
          </a>
        </button>
        <h1
          className="header"
        >
          Air Pollution
        </h1>
        <a
          aria-current="page"
          className="link text-light active"
          href="/"
          onClick={[Function]}
        >
          <svg
            aria-hidden="true"
            className="svg-inline--fa fa-arrows-rotate icon"
            data-icon="arrows-rotate"
            data-prefix="fas"
            focusable="false"
            role="img"
            style={Object {}}
            text="reload"
            viewBox="0 0 512 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M464 16c-17.67 0-32 14.31-32 32v74.09C392.1 66.52 327.4 32 256 32C161.5 32 78.59 92.34 49.58 182.2c-5.438 16.81 3.797 34.88 20.61 40.28c16.89 5.5 34.88-3.812 40.3-20.59C130.9 138.5 189.4 96 256 96c50.5 0 96.26 24.55 124.4 64H336c-17.67 0-32 14.31-32 32s14.33 32 32 32h128c17.67 0 32-14.31 32-32V48C496 30.31 481.7 16 464 16zM441.8 289.6c-16.92-5.438-34.88 3.812-40.3 20.59C381.1 373.5 322.6 416 256 416c-50.5 0-96.25-24.55-124.4-64H176c17.67 0 32-14.31 32-32s-14.33-32-32-32h-128c-17.67 0-32 14.31-32 32v144c0 17.69 14.33 32 32 32s32-14.31 32-32v-74.09C119.9 445.5 184.6 480 255.1 480c94.45 0 177.4-60.34 206.4-150.2C467.9 313 458.6 294.1 441.8 289.6z"
              fill="currentColor"
              style={Object {}}
            />
          </svg>
        </a>
      </div>
    </nav>
  </div>
  <div
    className="pollution m-5"
  >
    <h3
      className="text-white m-2"
    />
    <div
      className="m-2"
    >
      <img
        alt="flag"
        className="w-25 img"
      />
    </div>
    <p
      className="m-2 text-warning"
    >
      Air pollution ( μg/m3 ) stats - 2022:
    </p>
    <table
      className="table"
    >
      <thead>
        <tr
          className="table-light"
        >
          <th
            scope="col"
          >
            gas
          </th>
          <th
            scope="col"
          >
            μg/m3
          </th>
        </tr>
      </thead>
      <tbody>
        <tr
          className="table-secondary"
        >
          <th
            scope="row"
          >
            CO
          </th>
          <td />
        </tr>
        <tr
          className="table-danger"
        >
          <th
            scope="row"
          >
            NO
          </th>
          <td />
        </tr>
        <tr
          className="table-warning"
        >
          <th
            scope="row"
          >
            NO2
          </th>
          <td />
        </tr>
        <tr
          className="table-info"
        >
          <th
            scope="row"
          >
            NO2
          </th>
          <td />
        </tr>
        <tr
          className="table-info"
        >
          <th
            scope="row"
          />
          <td />
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;

exports[`Region List Region Lists renders correctly 1`] = `
<div>
  <div>
    <nav>
      <div
        className="d-flex justify-content-between align-items-center p-3 mb-2"
      >
        <button
          className="btn mt-2"
          type="button"
        >
          <a
            aria-current="page"
            className="link m-2 text-light active"
            href="/"
            onClick={[Function]}
          >
            <svg
              aria-hidden="true"
              className="svg-inline--fa fa-circle-arrow-left icon"
              data-icon="circle-arrow-left"
              data-prefix="fas"
              focusable="false"
              role="img"
              style={Object {}}
              viewBox="0 0 512 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M256 0C114.6 0 0 114.6 0 256c0 141.4 114.6 256 256 256s256-114.6 256-256C512 114.6 397.4 0 256 0zM384 288H205.3l49.38 49.38c12.5 12.5 12.5 32.75 0 45.25s-32.75 12.5-45.25 0L105.4 278.6C97.4 270.7 96 260.9 96 256c0-4.883 1.391-14.66 9.398-22.65l103.1-103.1c12.5-12.5 32.75-12.5 45.25 0s12.5 32.75 0 45.25L205.3 224H384c17.69 0 32 14.33 32 32S401.7 288 384 288z"
                fill="currentColor"
                style={Object {}}
              />
            </svg>
          </a>
        </button>
        <h1
          className="header"
        >
          Air Pollution
        </h1>
        <a
          aria-current="page"
          className="link text-light active"
          href="/"
          onClick={[Function]}
        >
          <svg
            aria-hidden="true"
            className="svg-inline--fa fa-arrows-rotate icon"
            data-icon="arrows-rotate"
            data-prefix="fas"
            focusable="false"
            role="img"
            style={Object {}}
            text="reload"
            viewBox="0 0 512 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M464 16c-17.67 0-32 14.31-32 32v74.09C392.1 66.52 327.4 32 256 32C161.5 32 78.59 92.34 49.58 182.2c-5.438 16.81 3.797 34.88 20.61 40.28c16.89 5.5 34.88-3.812 40.3-20.59C130.9 138.5 189.4 96 256 96c50.5 0 96.26 24.55 124.4 64H336c-17.67 0-32 14.31-32 32s14.33 32 32 32h128c17.67 0 32-14.31 32-32V48C496 30.31 481.7 16 464 16zM441.8 289.6c-16.92-5.438-34.88 3.812-40.3 20.59C381.1 373.5 322.6 416 256 416c-50.5 0-96.25-24.55-124.4-64H176c17.67 0 32-14.31 32-32s-14.33-32-32-32h-128c-17.67 0-32 14.31-32 32v144c0 17.69 14.33 32 32 32s32-14.31 32-32v-74.09C119.9 445.5 184.6 480 255.1 480c94.45 0 177.4-60.34 206.4-150.2C467.9 313 458.6 294.1 441.8 289.6z"
              fill="currentColor"
              style={Object {}}
            />
          </svg>
        </a>
      </div>
    </nav>
  </div>
  <div
    className="row m-4"
  >
    <div
      className="col-6 border"
    >
      <div
        className="region"
      >
        <button
          className="btn mt-2"
          onClick={[Function]}
          to="/countries"
          type="button"
        >
          <a
            className="link m-2 text-light"
            href="/countries"
            onClick={[Function]}
          >
            <div
              className="d-flex justify-content-center"
            >
              <img
                alt=""
                className="m-2 img"
                src="Asia.png"
              />
              <svg
                aria-hidden="true"
                className="svg-inline--fa fa-circle-arrow-right icon"
                data-icon="circle-arrow-right"
                data-prefix="fas"
                focusable="false"
                role="img"
                style={Object {}}
                viewBox="0 0 512 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M256 0C114.6 0 0 114.6 0 256c0 141.4 114.6 256 256 256s256-114.6 256-256C512 114.6 397.4 0 256 0zM406.6 278.6l-103.1 103.1c-12.5 12.5-32.75 12.5-45.25 0s-12.5-32.75 0-45.25L306.8 288H128C110.3 288 96 273.7 96 256s14.31-32 32-32h178.8l-49.38-49.38c-12.5-12.5-12.5-32.75 0-45.25s32.75-12.5 45.25 0l103.1 103.1C414.6 241.3 416 251.1 416 256C416 260.9 414.6 270.7 406.6 278.6z"
                  fill="currentColor"
                  style={Object {}}
                />
              </svg>
            </div>
            <div>
              <p>
                Asia
              </p>
              <p>
                50
                 
                Countries
              </p>
            </div>
          </a>
        </button>
      </div>
    </div>
    <div
      className="col-6 border"
    >
      <div
        className="region"
      >
        <button
          className="btn mt-2"
          onClick={[Function]}
          to="/countries"
          type="button"
        >
          <a
            className="link m-2 text-light"
            href="/countries"
            onClick={[Function]}
          >
            <div
              className="d-flex justify-content-center"
            >
              <img
                alt=""
                className="m-2 img"
                src="Americas.png"
              />
              <svg
                aria-hidden="true"
                className="svg-inline--fa fa-circle-arrow-right icon"
                data-icon="circle-arrow-right"
                data-prefix="fas"
                focusable="false"
                role="img"
                style={Object {}}
                viewBox="0 0 512 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M256 0C114.6 0 0 114.6 0 256c0 141.4 114.6 256 256 256s256-114.6 256-256C512 114.6 397.4 0 256 0zM406.6 278.6l-103.1 103.1c-12.5 12.5-32.75 12.5-45.25 0s-12.5-32.75 0-45.25L306.8 288H128C110.3 288 96 273.7 96 256s14.31-32 32-32h178.8l-49.38-49.38c-12.5-12.5-12.5-32.75 0-45.25s32.75-12.5 45.25 0l103.1 103.1C414.6 241.3 416 251.1 416 256C416 260.9 414.6 270.7 406.6 278.6z"
                  fill="currentColor"
                  style={Object {}}
                />
              </svg>
            </div>
            <div>
              <p>
                Americas
              </p>
              <p>
                56
                 
                Countries
              </p>
            </div>
          </a>
        </button>
      </div>
    </div>
    <div
      className="col-6 border"
    >
      <div
        className="region"
      >
        <button
          className="btn mt-2"
          onClick={[Function]}
          to="/countries"
          type="button"
        >
          <a
            className="link m-2 text-light"
            href="/countries"
            onClick={[Function]}
          >
            <div
              className="d-flex justify-content-center"
            >
              <img
                alt=""
                className="m-2 img"
                src="Oceania.png"
              />
              <svg
                aria-hidden="true"
                className="svg-inline--fa fa-circle-arrow-right icon"
                data-icon="circle-arrow-right"
                data-prefix="fas"
                focusable="false"
                role="img"
                style={Object {}}
                viewBox="0 0 512 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M256 0C114.6 0 0 114.6 0 256c0 141.4 114.6 256 256 256s256-114.6 256-256C512 114.6 397.4 0 256 0zM406.6 278.6l-103.1 103.1c-12.5 12.5-32.75 12.5-45.25 0s-12.5-32.75 0-45.25L306.8 288H128C110.3 288 96 273.7 96 256s14.31-32 32-32h178.8l-49.38-49.38c-12.5-12.5-12.5-32.75 0-45.25s32.75-12.5 45.25 0l103.1 103.1C414.6 241.3 416 251.1 416 256C416 260.9 414.6 270.7 406.6 278.6z"
                  fill="currentColor"
                  style={Object {}}
                />
              </svg>
            </div>
            <div>
              <p>
                Oceania
              </p>
              <p>
                27
                 
                Countries
              </p>
            </div>
          </a>
        </button>
      </div>
    </div>
    <div
      className="col-6 border"
    >
      <div
        className="region"
      >
        <button
          className="btn mt-2"
          onClick={[Function]}
          to="/countries"
          type="button"
        >
          <a
            className="link m-2 text-light"
            href="/countries"
            onClick={[Function]}
          >
            <div
              className="d-flex justify-content-center"
            >
              <img
                alt=""
                className="m-2 img"
                src="Antarctic.png"
              />
              <svg
                aria-hidden="true"
                className="svg-inline--fa fa-circle-arrow-right icon"
                data-icon="circle-arrow-right"
                data-prefix="fas"
                focusable="false"
                role="img"
                style={Object {}}
                viewBox="0 0 512 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M256 0C114.6 0 0 114.6 0 256c0 141.4 114.6 256 256 256s256-114.6 256-256C512 114.6 397.4 0 256 0zM406.6 278.6l-103.1 103.1c-12.5 12.5-32.75 12.5-45.25 0s-12.5-32.75 0-45.25L306.8 288H128C110.3 288 96 273.7 96 256s14.31-32 32-32h178.8l-49.38-49.38c-12.5-12.5-12.5-32.75 0-45.25s32.75-12.5 45.25 0l103.1 103.1C414.6 241.3 416 251.1 416 256C416 260.9 414.6 270.7 406.6 278.6z"
                  fill="currentColor"
                  style={Object {}}
                />
              </svg>
            </div>
            <div>
              <p>
                Antarctic
              </p>
              <p>
                5
                 
                Countries
              </p>
            </div>
          </a>
        </button>
      </div>
    </div>
    <div
      className="col-6 border"
    >
      <div
        className="region"
      >
        <button
          className="btn mt-2"
          onClick={[Function]}
          to="/countries"
          type="button"
        >
          <a
            className="link m-2 text-light"
            href="/countries"
            onClick={[Function]}
          >
            <div
              className="d-flex justify-content-center"
            >
              <img
                alt=""
                className="m-2 img"
                src="Europe.png"
              />
              <svg
                aria-hidden="true"
                className="svg-inline--fa fa-circle-arrow-right icon"
                data-icon="circle-arrow-right"
                data-prefix="fas"
                focusable="false"
                role="img"
                style={Object {}}
                viewBox="0 0 512 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M256 0C114.6 0 0 114.6 0 256c0 141.4 114.6 256 256 256s256-114.6 256-256C512 114.6 397.4 0 256 0zM406.6 278.6l-103.1 103.1c-12.5 12.5-32.75 12.5-45.25 0s-12.5-32.75 0-45.25L306.8 288H128C110.3 288 96 273.7 96 256s14.31-32 32-32h178.8l-49.38-49.38c-12.5-12.5-12.5-32.75 0-45.25s32.75-12.5 45.25 0l103.1 103.1C414.6 241.3 416 251.1 416 256C416 260.9 414.6 270.7 406.6 278.6z"
                  fill="currentColor"
                  style={Object {}}
                />
              </svg>
            </div>
            <div>
              <p>
                Europe
              </p>
              <p>
                53
                 
                Countries
              </p>
            </div>
          </a>
        </button>
      </div>
    </div>
    <div
      className="col-6 border"
    >
      <div
        className="region"
      >
        <button
          className="btn mt-2"
          onClick={[Function]}
          to="/countries"
          type="button"
        >
          <a
            className="link m-2 text-light"
            href="/countries"
            onClick={[Function]}
          >
            <div
              className="d-flex justify-content-center"
            >
              <img
                alt=""
                className="m-2 img"
                src="Africa.png"
              />
              <svg
                aria-hidden="true"
                className="svg-inline--fa fa-circle-arrow-right icon"
                data-icon="circle-arrow-right"
                data-prefix="fas"
                focusable="false"
                role="img"
                style={Object {}}
                viewBox="0 0 512 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M256 0C114.6 0 0 114.6 0 256c0 141.4 114.6 256 256 256s256-114.6 256-256C512 114.6 397.4 0 256 0zM406.6 278.6l-103.1 103.1c-12.5 12.5-32.75 12.5-45.25 0s-12.5-32.75 0-45.25L306.8 288H128C110.3 288 96 273.7 96 256s14.31-32 32-32h178.8l-49.38-49.38c-12.5-12.5-12.5-32.75 0-45.25s32.75-12.5 45.25 0l103.1 103.1C414.6 241.3 416 251.1 416 256C416 260.9 414.6 270.7 406.6 278.6z"
                  fill="currentColor"
                  style={Object {}}
                />
              </svg>
            </div>
            <div>
              <p>
                Africa
              </p>
              <p>
                59
                 
                Countries
              </p>
            </div>
          </a>
        </button>
      </div>
    </div>
  </div>
</div>
`;
